astropy/table/table.py

```diff
@@ -2319,8 +2319,14 @@ def copy(self, copy_data=True):
         copy_data : bool
             If `True` (the default), copy the underlying data array.
             Otherwise, use the same data array
+        max_rows : int
+            If None (the default), copy all the rows.
+            Otherwise copy only max_rows.
         '''
-        out = self.__class__(self, copy=copy_data)
+        if max_rows is not None:
+            out = self.__class__(self[:max_rows], copy=copy_data)
+        else:
```

Comment:
@taldcroft If ``max_rows`` entered is `greater` than ``len(table)`` than should an error be raised or default behavior of copying ``len(table)`` rows should be allowed?